Index: coremark-1.01/simple/core_portme.c
===================================================================
--- coremark-1.01.orig/simple/core_portme.c
+++ coremark-1.01/simple/core_portme.c
@@ -8,6 +8,7 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include "coremark.h"
+#include <pthread.h>
 
 #if VALIDATION_RUN
 	volatile ee_s32 seed1_volatile=0x3415;
@@ -90,7 +91,28 @@ secs_ret time_in_secs(CORE_TICKS ticks)
 	return retval;
 }
 
-ee_u32 default_num_contexts=1;
+ee_u32 default_num_contexts=4;
+#if (MULTITHREAD>1)
+ /* Function: core_start_parallel
+          Start benchmarking in a parallel context.
+
+         Three implementations are provided, one using pthreads, one using fork and shared mem, and one using fork and sockets.
+         Other implementations using MCAPI or other standards can easily be devised.
+ */
+ ee_u8 core_start_parallel(core_results *res) {
+         return (ee_u8)pthread_create(&(res->port.thread),NULL,iterate,(void *)res);
+ }
+ /* Function: core_stop_parallel
+         Stop a parallel context execution of coremark, and gather the results.
+
+         Three implementations are provided, one using pthreads, one using fork and shared mem, and one using fork and sockets.
+         Other implementations using MCAPI or other standards can easily be devised.
+ */
+ ee_u8 core_stop_parallel(core_results *res) {
+         void *retval;
+         return (ee_u8)pthread_join(res->port.thread,&retval);
+ }
+#endif
 
 /* Function : portable_init
 	Target specific initialization code 
Index: coremark-1.01/simple/core_portme.h
===================================================================
--- coremark-1.01.orig/simple/core_portme.h
+++ coremark-1.01/simple/core_portme.h
@@ -127,12 +127,16 @@ typedef size_t ee_size_t;
 	to fit a particular architecture. 
 */
 #ifndef MULTITHREAD
-#define MULTITHREAD 1
-#define USE_PTHREAD 0
+#define MULTITHREAD 4
+#define USE_PTHREAD 1
 #define USE_FORK 0
 #define USE_SOCKET 0
 #endif
+#if (MULTITHREAD>1)
+ #include <pthread.h>
+#define PARALLEL_METHOD "PThreads"
 
+#endif /* MULTITHREAD > 1 */
 /* Configuration : MAIN_HAS_NOARGC
 	Needed if platform does not support getting arguments to main. 
 	
@@ -164,6 +168,9 @@ typedef size_t ee_size_t;
 extern ee_u32 default_num_contexts;
 
 typedef struct CORE_PORTABLE_S {
+#if (MULTITHREAD>1)
+        pthread_t thread;
+#endif /* MULTITHREAD>1 */
 	ee_u8	portable_id;
 } core_portable;
 
Index: coremark-1.01/simple/core_portme.mak
===================================================================
--- coremark-1.01.orig/simple/core_portme.mak
+++ coremark-1.01/simple/core_portme.mak
@@ -10,7 +10,7 @@ CC = gcc
 #	Use this flag to define compiler options. Note, you can add compiler options from the command line using XCFLAGS="other flags"
 PORT_CFLAGS = -O3
 FLAGS_STR = "$(PORT_CFLAGS) $(XCFLAGS) $(XLFLAGS) $(LFLAGS_END)"
-CFLAGS = $(PORT_CFLAGS) -I$(PORT_DIR) -I. -DFLAGS_STR=\"$(FLAGS_STR)\"
+CFLAGS = $(PORT_CFLAGS) -pthread -DMULTITHREAD=4 -DUSE_PTHREAD -I$(PORT_DIR) -I. -DFLAGS_STR=\"$(FLAGS_STR)\"
 #Flag : LFLAGS_END
 #	Define any libraries needed for linking or other flags that should come at the end of the link line (e.g. linker scripts). 
 #	Note : On certain platforms, the default clock_gettime implementation is supported but requires linking of librt.
